zone-setup/src/bin/zone-setup.rs

```diff
@@ -560,6 +560,17 @@ async fn common_nw_set_up(
     Ipadm::set_interface_mtu(&datalink)
         .with_context(|| format!("failed to set MTU on datalink {datalink}"))?;
 
+    info!(
+        log, "Setting TCP recv_buf size to 1 MB";
+    );
+    Ipadm::set_tcp_recv_buf().context("failed to set TCP recv_buf")?;
+
+    info!(
+        log, "Setting TCP congestion control algorithm to cubic";
+    );
+    Ipadm::set_tcp_congestion_control()
+        .context("failed to set TCP congestion_control")?;
+
```

Comment:
Why is this being set to all zones and not just nexus?

Reply:
Because it's easier and Josh said it was "probably fine." However, it would probably be pretty easy to make it part of a command or flag that is only used by the Nexus zone setter upper.

Reply:
I think it's basically fine everywhere, FWIW.  We should test to make sure that everything still works before and after obviously but I don't expect there will be any issues

Reply:
Sadly, it wouldn't be as straightforward as it seems to implement only to Nexus. Currently there's no nexus-specific SMF service that's required before the nexus service starts. We'd have to create a new service and do something like https://github.com/oxidecomputer/omicron/pull/5440.

I'm a bit wary of doing a blanket change to system defaults of all zones as we don't really have a comprehensive integration testing suite for this.

If everyone is OK with it, perhaps we can go the safer way? Not sure how urgent this is.

Reply:
But also I think it's low risk and I would probably just do it

Reply:
I trust the duke of illumos on this. From what I read [elsewhere](https://www.illumos.org/issues/15019), it sounds like the default value is probably low for modern hardware anyway!

Reply:
This service also runs in the switch zone, are we sure this wouldn't have unintended consequences?

Reply:
I'll leave it up to all of you! If you're all ok with it then I'm ok with it. Can we just add a comment on the code with an explanation of why we're setting those new values?

Reply:
I've added a comment explaining why we're increasing `recv_buf`.